kernel/src/kcmdline.rs

```diff
@@ -38,23 +41,39 @@ pub enum ModuleArg {
 #[derive(Debug)]
 pub struct KCmdlineArg {
     initproc: InitprocArgs,
+    console_names: Vec<String>,
     module_args: BTreeMap<String, Vec<ModuleArg>>,
 }
 
 // Define get APIs.
 impl KCmdlineArg {
+    /// Gets the singleton instance of `KCmdlineArg`.
+    pub fn singleton() -> &'static KCmdlineArg {
+        static INSTANCE: Once<KCmdlineArg> = Once::new();
+
+        INSTANCE.call_once(|| KCmdlineArg::from(boot_info().kernel_cmdline.as_str()))
+    }
+
     /// Gets the path of the initprocess.
     pub fn get_initproc_path(&self) -> Option<&str> {
         self.initproc.path.as_deref()
     }
+
     /// Gets the argument vector(argv) of the initprocess.
     pub fn get_initproc_argv(&self) -> &Vec<CString> {
         &self.initproc.argv
     }
+
     /// Gets the environment vector(envp) of the initprocess.
     pub fn get_initproc_envp(&self) -> &Vec<CString> {
         &self.initproc.envp
     }
+
+    /// Gets the list of system console names specified in the command line.
```

Comment:
`list` -> `vector`. It is slightly more correct and consistent with other doc comments here.